pyvista/core/utilities/points.py

```diff
@@ -459,3 +464,156 @@ def vector_poly_data(orig, vec):
+        A more memory-efficient computation may be supported in a future release.
+
+    .. versionadded:: 0.45.0
+
+    Parameters
+    ----------
+    points : MatrixLike[float]
+        Nx3 array of points.
+
+    return_std : bool, default: False
+        If ``True``, also returns the standard deviation of the points along each axis.
+        Standard deviation is computed as the square root of the eigenvalues of the
+        mean-centered covariance matrix, divided by the number of points.
+
+    Returns
+    -------
+    numpy.ndarray
+        3x3 orthonormal array with the principal axes as row vectors.
+
+    numpy.ndarray
+        Three-item array of the axis sizes.
+
+    Examples
+    --------
+    >>> import pyvista as pv
+    >>> import numpy as np
+    >>> rng = np.random.default_rng(seed=0)
+
+    Create a mesh with points that have the largest variation in ``X``,
+    followed by ``Y``, then ``Z``.
+
+    >>> radii = np.array((6, 3, 1))  # x-y-z radii
+    >>> mesh = pv.ParametricEllipsoid(
+    ...     xradius=radii[0], yradius=radii[1], zradius=radii[2]
+    ... )
+
+    Plot the mesh and highlight its points in black.
+
+    >>> p = pv.Plotter()
+    >>> _ = p.add_mesh(mesh)
+    >>> _ = p.add_points(mesh, color='black')
+    >>> _ = p.show_grid()
+    >>> p.show()
+
+    Compute its principal axes and return the standard deviation.
+
+    >>> axes, std = pv.principal_axes(mesh.points, return_std=True)
+    >>> axes
+    pyvista_ndarray([[-1.0000000e+00, -3.8287229e-08,  3.6589407e-10],
+                     [-3.8287229e-08,  1.0000000e+00, -3.0685656e-09],
+                     [-3.6589393e-10, -3.0685656e-09, -1.0000000e+00]],
+                    dtype=float32)
+
+    Note that the principal axes have ones along the diagonal and zeros
+    in the off-diagonal. This indicates that the first principal axis is
+    aligned with the x-axis, the second with the y-axis, and third with
+    the z-axis. This is expected, since the mesh is already axis-aligned.
+
+    However, since the signs of the principal axes are arbitrary, the
+    first and third axes in this case have a negative direction.
+
+    Show the standard deviation along each axis.
+
+    >>> std
+    array([0.03060594, 0.01530297, 0.00714213], dtype=float32)
+
+    Convert the values to proportions for analysis.
+
+    >>> std / sum(std)
+    array([0.5769149 , 0.28845745, 0.1346276 ], dtype=float32)
+
+    From this result, we can determine that the axes explain approximately
+    58%, 29%, and 13% of the total variance in the points, respectively.
+
+    Let's compare this to the proportions of the known radii of the ellipsoid.
+
+    >>> radii / sum(radii)
+    array([0.6, 0.3, 0.1])
+
+    Note how the two ratios are similar, but do not match exactly. This is
+    because the points of the ellipsoid are prolate and are denser near the
+    poles. If the points were normally distributed, however, the proportions
+    would match exactly.
```

Comment:
Your ellipsoid is _oblate_, so I think there's a competition between "smaller axis along z" and "denser points along z". It might not be obvious which one wins. But this is very handwavy, based on my earlier surprises with `pv.Sphere()`...

Further musing: I'm thinking that (according to my current newly found understanding) PCA fits the covariance of a distribution with normals, a large issue here is that the distribution is very "not normal". My hunch is that we'd get better agreement with a space-filling ellipsoid cloud. But I don't think it would be worth anyone's time looking into that :smile:

Reply:
I'm not really familiar with this technical nuance, so I'll leave it up to you, feel free to modify the wording or the example itself, or even remove it altogether if you don't think it adds value.

TBH I feel the real value of this method will come through with examples using real datasets and applying transformations with the principal axes. I plan on adding a full example in a separate PR for this.